devito/types/grid.py

```diff
@@ -258,36 +267,69 @@ def is_distributed(self, dim):
         """True if ``dim`` is a distributed Dimension, False otherwise."""
         return any(dim is d for d in self.distributor.dimensions)
 
-    @property
-    def _const(self):
-        """The type to be used to create constant symbols."""
-        return Constant
-
-    def _make_stepping_dim(self, time_dim, name=None):
-        """Create a SteppingDimension for this Grid."""
-        if name is None:
-            name = '%s_s' % time_dim.name
-        return SteppingDimension(name=name, parent=time_dim)
-
-    def _make_time_dim(self, spacing):
-        """Create a TimeDimension for this Grid."""
-        return TimeDimension(name='time', spacing=spacing)
+    @cached_property
+    def _arg_names(self):
+        ret = []
+        ret.append(self.time_dim.spacing.name)
+        ret.extend([i.name for i in self.origin_map])
+        for i in self.spacing_map:
+            try:
+                ret.append(i.name)
+            except AttributeError:
+                # E.g., {n*h_x: v} (the case of ConditionalDimension)
+                ret.extend([a.name for a in i.free_symbols])
+        return tuple(ret)
 
     @memoized_meth
     def _arg_defaults(self):
         """A map of default argument values defined by this Grid."""
         args = ReducerMap()
 
+        # Dimensions size
         for k, v in self.dimension_map.items():
             args.update(k._arg_defaults(_min=0, size=v.loc))
 
+        # Dimensions spacing
+        try:
+            args.update({k.name: v for k, v in self.spacing_map.items()})
+        except AttributeError:
+            # See issue #1524
+            # We check whether we're in the special case {n*h_x: v, ...}, which is
```

Comment:
Wouldnt it be "easier" to fix the map so that it returns `{h_x: v/n, ...}`?

Reply:
I actually totally agree with you. I've just changed it to do as you suggest